### DISCUSSION TITLE
TypeScript unable to serialize loader data with "defer" and an early return

### DISCUSSION THREAD

**Discussion by silvenon** [remix-run/remix#5239] [Q&A] [answered]
I'm trying out `defer`, but my loader has mutliple returns, and in these cases TypeScript fails to serialize the data, unlike when using `json`.

```tsx
export function loader(args: LoaderArgs) {
  if (!args.params.slug) return redirect('/blog')
  return defer({
    title: 'My blog post'
  })
}

export default function Post() {
  const data = useLoaderData<typeof loader>()
  return (
    // Property 'title' does not exist on type (…)
    <h1>{data.title}</h1>
  )
}
```

Am I doing something wrong?

**Comment by machour** [comment, marked as answer]
Seeing the same thing on my side
(note: you could throw the redirect as a workaround)

**Reply by silvenon** [reply]
Thanks for the workaround! ❤️

**Reply by silvenon** [reply]
I'll mark this as the answer, I should always lean towards throwing redirects in these types of situations, and it's less work for TypeScript to figure out.

**Comment by ebz-jdev** [comment]
Seeing this issue as well

**Reply by silvenon** [reply]
Does @machour's answer solve your problem? https://github.com/remix-run/remix/discussions/5239#discussioncomment-4769936

It did for me.
